capstone_project/src/main/webapp/scripts/mapsScriptTemplate.js

```diff
@@ -22,6 +22,54 @@ function createMap() { // eslint-disable-line no-unused-vars
 
   // Create a marker for each school and load these onto the map.
   loadMarkersOntoMap(map);
+
+  // Create the autocomplete Places search.
+  loadAutocomplete(map);
+}
+
+/** Creates and displays an autocompleted Places search engine on the page.
+ * @param {Object} map - A google maps Map object.
+*/
+function loadAutocomplete(map) {
+  // Set up the autocomplete search widget.
+  const input = document.getElementById('school-search');
+  const autocomplete = new google.maps.places.Autocomplete(input);
+  autocomplete.bindTo('bounds', map);
+  autocomplete.setFields(
+      ['address_components', 'geometry', 'icon', 'name']);
+
+  // Grab the place submission from autocomplete and post it only when the
+  // 'submit' button is clicked.
+  document.getElementById('school-submit')
+      .addEventListener('click', function() {
+        const place = autocomplete.getPlace();
+        if (!place.geometry) {
+          // The user entered the name of a Place that was not suggested,
+          // or the request for Place Details failed.
+          window.alert(`No details were found for submission: '${place.name}'`);
+          return;
+        }
+        const submission = {
+          name: place.name,
+          latitude: place.geometry.location.lat(),
+          longitude: place.geometry.location.lng(),
+        };
+        postSchool(submission);
+      });
+}
+
+/** Sends the school submission to a MapServlet.
+ * @param {Object} submission - An object representing a school submission.
+*/
+function postSchool(submission) {
+  fetch('/school-data', {
+    method: 'POST',
+    body: JSON.stringify(submission),
+    headers: {
+      'Accept': 'application/json',
+    },
+  });
+  location.reload();
```

Comment:
would this cause the browser to reload before the fetch function is complete, since fetch is asynchronous? I would call .then on fetch so something like:
```
fetch(...).then((response) => location.reload());
``` 
that way a reload happens after a school is finished uploading

Reply:
sounds good, I just fixed it! @Leoism